src/main/java/org/terasology/combatSystem/weaponFeatures/systems/AttackSystem.java

```diff
@@ -10,18 +10,35 @@
 import org.terasology.entitySystem.systems.BaseComponentSystem;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.characters.CharacterImpulseEvent;
 import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.location.LocationComponent;
+import org.terasology.math.geom.Vector3f;
 import org.terasology.world.block.items.BlockItemComponent;
 import org.terasology.world.block.items.OnBlockItemPlaced;
 
 @RegisterSystem
-public class AttackSystem extends BaseComponentSystem{
-    
-    @ReceiveEvent( components = {PrimaryAttackComponent.class})
-    public void primaryAttack(ActivateEvent event, EntityRef entity){
+public class AttackSystem extends BaseComponentSystem {
+
+    @ReceiveEvent(components = {PrimaryAttackComponent.class})
+    public void primaryAttack(ActivateEvent event, EntityRef entity) {
         entity.send(new PrimaryAttackEvent(event));
     }
-    
+
+    @ReceiveEvent( components = {PrimaryAttackComponent.class})
```

Comment:
This line encodes that the event handler will only trigger if the `entity` has a `PrimaryAttackComponent`, that's what `components = {PrimaryAttackComponent.class}` means.

I don't think this is necessary, and the annotation can simply be the following:

```suggestion
    @ReceiveEvent
```